sources/imdsclient/src/lib.rs

```diff
@@ -103,6 +103,27 @@ impl ImdsClient {
         self.fetch_string(&node_ip_target).await
     }
 
+    /// Gets the IPV6 addresses from instance metadata.
+    pub async fn fetch_ipv6_addresses(&mut self) -> Result<Option<Vec<String>>> {
+        // Get the mac address for the primary network interface
```

Comment:
The interface here is a bit awkward since we assume only the first (primary) interface is interesting, even though the name implies that all IPv6s will be returned to the caller.

It might be better to name it `fetch_primary_ipv6_address` and then do both of the `first` filtering operations. That way we're also centralizing the logic of figuring out which IPv6 address is the "main" one, which doesn't seem all that well documented and therefore might not be implemented the same way by every caller.